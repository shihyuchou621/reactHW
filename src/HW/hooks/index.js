import HW1 from './HW1';
import HW2 from './HW2';
import HW3 from './HW3';
import HW4 from './HW4';
import HW5 from './HW5';
import HW6 from './HW6';
import HW7 from './HW7';
import HW8 from './HW8';
import HW9 from './HW9';
import HW10 from './HW10';
import HW11 from './HW11';
import HW12 from './HW12';
import HW13 from './HW13';
import HW14 from './HW14';
import HW15 from './HW15';
import HW16 from './HW16';
import HW17 from './HW17';
import HW18 from './HW18';
import HW19 from './HW19';
import HW20 from './HW20';
import HW21 from './HW21';
import HW22 from './HW22';
import HW23 from './HW23';
import HW25 from './HW25';
import HW26 from './HW26';
import HW27 from './HW27';

export default {
  HW1,
  HW2,
  HW3,
  HW4,
  HW5,
  HW6,
  HW7,
  HW8,
  HW9,
  HW10,
  HW11,
  HW12,
  HW13,
  HW14,
  HW15,
  HW16,
  HW17,
  HW18,
  HW19,
  HW20,
  HW21,
  HW22,
  HW23,
  HW25,
  HW26,
  HW27,
};