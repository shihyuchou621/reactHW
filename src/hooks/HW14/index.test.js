import React from "react";

import {
  cleanup,
  render,
  fireEvent,
  wait,
} from "@testing-library/react";

// import { waitFor } from "@testing-library/dom";

import HW from ".";

afterEach(cleanup);

const mockFetch = (data) =>
  jest.fn(() =>
    Promise.resolve({
      json: () => Promise.resolve(data),
    })
  );

global.fetch = mockFetch({
  count: 1050,
  next: "https://pokeapi.co/api/v2/pokemon/?offset=3&limit=3",
  previous: null,
  results: [
    { name: "bulbasaur", url: "https://pokeapi.co/api/v2/pokemon/1/" },
    { name: "ivysaur", url: "https://pokeapi.co/api/v2/pokemon/2/" },
    { name: "venusaur", url: "https://pokeapi.co/api/v2/pokemon/3/" },
  ],
});

test('should fetch data and get pokemon', async () => {
  const { container, getByTestId } = render(<HW />);
  fireEvent.change(getByTestId("input"), { target: { value: 'bulb'}});

  await wait(() => {
    expect(getByTestId("bulbasaur")).toHaveTextContent("bulbasaur");
    expect(container).toMatchSnapshot();
  });
});
