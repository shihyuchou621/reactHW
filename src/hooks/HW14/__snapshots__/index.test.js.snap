// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should fetch data and get pokemon 1`] = `
<div>
  <div
    class="container HW14"
  >
    <div
      class="input-group mb-3"
    >
      <div
        class="input-group-prepend"
      >
        <span
          class="input-group-text"
          id="basic-addon1"
        >
          keyword
        </span>
      </div>
      <input
        class="form-control"
        data-testid="input"
        placeholder="name..."
        value="bulb"
      />
    </div>
    <div
      class="inline"
    >
      <img
        alt="bulbasaur"
        src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
      />
      <div
        data-testid="bulbasaur"
      >
        bulbasaur
      </div>
    </div>
  </div>
</div>
`;
